### ISSUE TITLE
cast error on windows os_version, and missing build revision?

### ISSUE THREAD

**Issue by directionless** [osquery/osquery#6082]
# Bug report

We noticed a couple of issues with the windows version table.

Windows seems to have a major, minor, build, and build revision. Notably the is no `patch` level. This means that when we select against the `os_version` table, we get cast errors:

```
osquery> SELECT patch FROM os_version;                                                                                                        
I1125 20:47:07.555104  3564 dynamic_table_row.cpp:114] Error casting patch () to INTEGER                                                      
patch =                                                                                                                                       
```

Also, it could be interesting to add a build revision. I know this is in the registry, not sure it's in WMI
```
osquery> select data from registry WHERE key = "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\" and name = "UBR" ;          
 data = 864                                                                                                                                   
```

### What operating system and version are you using?

```
osquery> SELECT version, build, platform FROM os_version;                                                                                     
 version = 10.0.17763                                                                                                                         
   build = 17763                                                                                                                              
platform = windows                                                                                                                            
```

### What version of osquery are you using?
```
osquery> SELECT version from osquery_info;                                                                                                    
version = 4.1.1                                                                                                                               
```

**Comment by mike2523** [comment]
revision from UBR hive is one place, another we use is for subversion is:

```
osquery> select * from kernel_info;
+----------------+---------------------------------------------------------------------+----------------------------------+---------------------------------------------------+
| version        | arguments                                                           | path                             | device                                            |
+----------------+---------------------------------------------------------------------+----------------------------------+---------------------------------------------------+
| 10.0.18362.476 |  NOEXECUTE=OPTIN  HYPERVISORLAUNCHTYPE=AUTO  FVEBOOT=2125824  NOVGA | C:\WINDOWS\System32\ntoskrnl.exe | \\?\Volume{7e11dfe2-43b6-4ee7-9081-ceb37fb08c39}\ |
+----------------+---------------------------------------------------------------------+----------------------------------+---------------------------------------------------+
osquery>
```

**Comment by theopolis** [comment]
Ok we can fix this by adding `0` to `patch`.

**Comment by directionless** [comment]
Do you think we should add a `0`, or not cast on windows and leave it null? I couldn't decide, so didn't PR anything

**Comment by theopolis** [comment]
We could also stop trying to cast empty/null columns.

**Comment by directionless** [comment]
fixed in #6371
